### ISSUE TITLE
`DotDotDotToken` is parsed in JSX attribute values

### ISSUE THREAD

**Issue by Andarist** [microsoft/TypeScript#55498]
### 🔎 Search Terms

JSX spread dotdotdot parse jsx attribute value

### 🕗 Version & Regression Information

- This is the behavior in every version I tried

### ⏯ Playground Link

https://www.typescriptlang.org/play?ts=5.3.0-dev.20230824#code/CYUwxgNghgTiAEYD2A7AzgF3gJRFMGAXPFCgJ4BQoksCKUAtiGgA74IBSAygBrwDeFeMPgBLFBhAwAZu3gBJCTHFpRYAKIQQTCWgFCRhgNoBrYpmUoA5gF1ipSoYC+FFxWTosPe+XepMJD6UFAA8fFAAvPwAdLFQTvAA9AB8QA

### 💻 Code

```ts
declare const React: any
declare namespace JSX {
    interface IntrinsicElements {
        [k: string]: any
    }
}

const X: any
const a: any

<X a={...a} />
```

### 🙁 Actual behavior

It gets parsed correctly

### 🙂 Expected behavior

It shouldn't parse or should be raised as an error at a later stage. 

### Additional information about the issue

This doesn't make sense since this is roughly equivalent to:
```ts
X({
  a: ...a
})
```

It's a parse error in:
- [Babel](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=DwDQBAhgvA3gdAiBfMB6AfEA&debug=false&forceAllTransforms=true&modules=false&shippedProposals=false&circleciRepo=&evaluate=true&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact&prettier=false&targets=&version=7.22.11&externalPlugins=%40babel%2Fplugin-proposal-explicit-resource-management%407.20.0&assumptions=%7B%7D)
- [Flow](https://flow.org/try/#1N4Igxg9gdgZglgcxALlAIwIZoKYBsD6uEEAztvhgE6UYCe+JADpdhgCYowa5kA0I2KAFcAtiRQAXSkOz9sADwxgJ+NPTbYuQ3BMnTZA+Y2yU4IwRO4A6SFBIrGVDGM7c+IFkolXpUCWewUEAwhCQgRDH8wEH4hMnwROHlsNnw4KHwwSLAAC3wANyo4LFxscWQuHgMNZmwsiRSAWglaY1cq-hIAa2wJXNpG4Vxcdvdu3v7B0RxKUYMhKDBSqmbWwIq3eagoOrKSKgH0wtMMPznY7d2SfcoBiEZ-aG5G3Ix085AF-ZhsRoRehqUEiNMgSQHlSruBZxJrMcJwMhzAC+-EgGiCtnsAAIABrILGnWgAHUW0GxGCxAF4sQBtAC6JJJAB4cQTKcArJyMEisQB6AB8MRA+RMJDg0CC+QADFYAEwARgArFZ5SAkUA) (but maybe that playground uses Babel under the hood?)
- [esbuild](https://esbuild.github.io/try/#dAAwLjE5LjIALS1sb2FkZXI9anN4ADxYIGE9ey4uLmF9IC8+)
